### PR TITLE
Do not submit enhancements

### PR DESCRIPTION
**Pull request by jbohanon**
# Description

Minor updates to `do-not-submit`:
- check .proto files
- don't attempt to check deleted files

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/main/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ci/do-not-submit.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ DO_NOT_SUBMIT_REGEX="[[:space:]]*//[[:space:]]*DO_NOT_SUBMIT"
 file_count=0
 for filename in "$@"; do
   # Only check *.go files for now
-  if [[ "$filename" == *".go" || "$filename" == *"go.mod" ]]; then
+  if [[ "$filename" == *".go" || "$filename" == *"go.mod" || "$filename" == *".proto" ]]; then
```
Comment (sam-heilbron) [inline ci/do-not-submit.sh:25]: nit: the comment above this line indicates that only go files are checked, but this is no longer true

Comment (sam-heilbron) [reply ci/do-not-submit.sh:25]: There's also a comment at the top of the file

**Review by sam-heilbron — COMMENTED** [review]
I'm good with this as is, and happy to approve. There are 2 small comments that are now out of date with this change

**Review by inFocus7 — APPROVED** [review]
🚀

**Review by jenshu — APPROVED** [review]

**Pull request merged**